File: frontend/src/app/shared/journal-template.ts
export class JournalTemplate {
  constructor(
    public userId: number,
    public name: string,
    public version: number,
    public content: Question [],
    public id?: string
  ) {
  }
}

export class Question {
  constructor(
    public id: string,
    public label: string,
    public required: boolean = false
  ) {
  }
}

File: spring-boot-app/src/main/kotlin/org/example/jobjournaler/entity/JournalTemplate.kt
package org.example.jobjournaler.entity

import org.springframework.data.annotation.Id
import org.springframework.data.relational.core.mapping.Table

@Table(name = "journal_template")
data class JournalTemplate(
    @Id
    val id: String?,
    val userId: Long,
    val name: String,
    var version: Int?,
    var content: String,
)
